### PR TITLE
Jwt Auth

### PR DESCRIPTION
**Pull request by aqeelmozumder**
Latest Update:

Only one package jsonwebtoken is added, express-session has been removed

Make Sure to put the header in the connection settings as header: `Authorization` value:` {{token}}`

Creates only one token now named as accessToken

Flow of Auth:
User logs in
Generates accesstoken which is set to expire in 30 mins
All resolvers verifies token from the header

### CODE REVIEW

File: src/context/index.ts [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
 import { request } from '../algorithm';
 
 export interface Context {
-  // TODO: keep here until we have login implemented. a partial given it could be undefined
-  session: Session & Partial<SessionData>;
-  // TODO: add context (aka. prisma, auth, etc.)
-  logout: () => Promise<void>;
-  // dependencies
+  req: any;
+  res: any;
```
Comment (aomi) [inline src/context/index.ts:6]: We shouldn't expose these into the context. If you need access you a `user` property define it here instead and populate it within `createContext`

File: src/context/index.ts [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
   algorithm: typeof request;
 }
 
 export async function createContext({
   req,
+  res,
 }: {
   req: Request;
+  res: any;
 }): Promise<Context> {
```
Comment (aomi) [inline src/context/index.ts:16]: Under here I think it makes sense to check for the request headers and verify the token and fetch user details etc.

Comment (aomi) [reply src/context/index.ts:16]: You could even nest it into a `session` object so you can keep all existing code

File: src/index.ts [hunk 3]
```diff
@@ -42,6 +37,35 @@ async function start(app: Express, typeDefs: any, resolvers: Resolvers) {
...
+    try {
+      // Verifying Refresh Token
+      await tokenRefresh(refreshToken, res, req, next);
+    } catch (e) {
+      // Refresh Token Expired
+      console.log('TOKENS Have Expired, Please Login again');
+    }
+
+    next();
+  });
```
Comment (aomi) [inline src/index.ts:67]: Move this into `createContext` imo. We don't have any endpoints needing auth that is REST-based anyway.

File: src/context/index.ts (hunk 2 above)
Comment (aqeelmozumder) [reply src/context/index.ts:13]: I still have to pass req

File: src/auth/index.ts [hunk 4]
```diff
@@ -53,20 +55,27 @@ async function login(
...
     };
   } else {
-    ctx.session.user = user;
+    const accessToken = sign(
+      {
+        username: user.username,
+      },
+      SECRET_ACCESSTOKEN,
+      { expiresIn: '30m' }
+    ); // token expires in 30 mins
```
Comment (jayden-chan) [inline src/auth/index.ts:67]: This seems annoyingly low. For the purposes of this project it's probably fine to set this to something pretty long like 48 hours tbh. Will cut down on annoyances during development and during demos

**Review by MustafaWasif — APPROVED** [review]

**Review by aomi — APPROVED** [review]

**Pull request merged**